index.js

```diff
@@ -99,27 +104,35 @@ exports.kinesisHandler = function (records, context, callback) {
       logger.info({'message': 'Posting...'})
       logger.info({'message': 'Response: ' + response.statusCode})
       if (response.statusCode !== 200) {
-        if (response.statusCode === 401) {
-          // Clear access token so new one will be requested on retried request
-          CACHE['accessToken'] = null
-        }
-
         callback(new Error())
         logger.error({'message': 'POST Error! ', 'response': response})
         return
       }
+      logger.info({'message': 'POST Success'})
+    })
+  }
+
 
-      if (error) {
+  function deleteRecords(record){
+    logger.info({'message': 'Deleting records'})
+    var options = {
+      uri: process.env['MLN_API_URL'] + '/teacher_set',
+      method: 'DELETE',
+      // MLN application currently does not require NYPL OAUTH Authentication 
+      //headers: { Authorization: `Bearer ${accessToken}` },
+      body: records,
+      json: true
+    }
+    request(options, function (error, response, body) {
+      logger.info({'message': 'Deleting...'})
+      logger.info({'message': 'Response: ' + response.statusCode})
+      if (response.statusCode !== 200) {
         callback(new Error())
-        logger.error({'message': 'POST Error! ', 'error': error})
+        logger.error({'message': 'DELETE Error! ', 'response': response})
```

Comment:
Let's log before the callback.  Just in case, always safer to log first, functionality next.

Reply:
That sounds redundant if the callback is going to be executed anyway and already include's a log statement. What would be the importance of this new log statement?